cmake/BDWGC.cmake

```diff
@@ -0,0 +1,74 @@
+macro(p4c_obtain_bdwgc)
+  include(CheckSymbolExists)
+
+  option(
+    P4C_USE_PREINSTALLED_BDWGC
+    "Look for a preinstalled version of BDWGC in the system instead of installing a prebuilt binary using FetchContent."
+    OFF
+  )
+  set(LIBGC_LIBRARIES gc gccpp cord)
+  set(HAVE_LIBGC 1)
+
+  if(P4C_USE_PREINSTALLED_BDWGC)
+    OPTION (ENABLE_MULTITHREAD "Use multithreading" OFF)
+    find_package(LibGc 7.2.0 REQUIRED)
+    check_function_exists (GC_print_stats HAVE_GC_PRINT_STATS)
+  else()
+    # Print out download state while setting up BDWGC.
+    set(FETCHCONTENT_QUIET_PREV ${FETCHCONTENT_QUIET})
+    set(BUILD_SHARED_LIBS_PREV ${BUILD_SHARED_LIBS})
+    set(FETCHCONTENT_QUIET OFF)
+    set(BUILD_SHARED_LIBS OFF CACHE BOOL "")
+
+    # Turn on C++ support.
+    set(enable_cplusplus ON CACHE BOOL "C++ support")
+    set(install_headers OFF CACHE BOOL "Install header and pkg-config metadata files.")
+    set(enable_docs ${ENABLE_DOCS} CACHE BOOL "Build and install documentation.")
+    # We may have to compile very large programs which allocate a lot on the heap.
+    set(enable_large_config ON CACHE BOOL "Optimize for large heap or root set.")
+    set(without_libatomic_ops OFF CACHE BOOL "Use atomic_ops.h in libatomic_ops/src")
+    # Redirect all malloc calls in the program.
+    set(enable_redirect_malloc OFF CACHE BOOL "Redirect malloc and friends to GC routines.")
```

Comment:
We do this manually (overriding malloc to call GC_malloc) in our lib/gc.cpp -- perhaps by turning this on we could avoid that?

Reply:
I tried that previously but I saw some performance regressions. Maybe we can experiment with this once this PR has been merged.